lib/web/machineid_test.go

```diff
@@ -296,3 +296,37 @@ func TestGetBotByName(t *testing.T) {
 	_, err = pack.clt.Get(ctx, fmt.Sprintf("%s/%s", endpoint, "invalid-bot"), nil)
 	require.Error(t, err)
 }
+
+func TestEditBot(t *testing.T) {
+	ctx := context.Background()
+	env := newWebPack(t, 1)
+	proxy := env.proxies[0]
+	pack := proxy.authPack(t, "admin", []types.Role{services.NewPresetEditorRole()})
+	clusterName := env.server.ClusterName()
+	endpoint := pack.clt.Endpoint(
+		"webapi",
+		"sites",
+		clusterName,
+		"machine-id",
+		"bot",
+	)
+
+	// create a bot named `test-bot-edit`
+	botName := "test-bot-edit"
+	_, err := pack.clt.PostJSON(ctx, endpoint, CreateBotRequest{
+		BotName: botName,
+		Roles:   []string{""},
```

Comment:
Welp - I'm fairly sure it shouldn't be possible to create a bot with a role which is just an empty string... I'll add it to my to-do list. I'd avoid the test depending on this being possible as it'll break when I fix it.